source/common/http/utility.cc

```diff
@@ -527,7 +527,13 @@ void Utility::sendLocalReply(const bool& is_reset, const EncodeFunctions& encode
 
   // Respond with a gRPC trailers-only response if the request is gRPC
   if (local_reply_data.is_grpc_) {
-    response_headers->setStatus(std::to_string(enumToInt(Code::OK)));
+    // If configured, the localReply HTTP status code will be the one from the
+    // called service's response headers. Default 'OK' reflects successful
+    // gRPC call. https://github.com/envoyproxy/envoy/issues/11079
```

Comment:
I don't think this is correct. According to the grpc spec, grpc errors are communicated through the grpc code, and http-200 should be present in the response even on failure.

Reply:
Are you saying you don't think the comment is correct, or the change to make the HTTP code match the DENIED response?

Reply:
I'm saying that according to the gRPC spec, both success and failure of the gRPC call should have http-200 as the response code, so the comment `Default 'OK' reflects successful gRPC call` isn't correct, because `OK` doesn't mean it succeeded.

Reply:
Thanks for clarifying, I'll correct the text.